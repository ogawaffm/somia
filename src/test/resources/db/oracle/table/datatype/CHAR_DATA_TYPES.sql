BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.CHAR_DATA_TYPES PURGE';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.CHAR_DATA_TYPES
(

    -- with default size of 1 character
    CHAR_COL                                  CHAR,
    CHARACTER_COL                             CHARACTER,

    NCHAR_COL                                 NCHAR,
    NATIONAL_CHAR_COL                         NATIONAL CHAR,
    NATIONAL_CHARACTER_COL                    NATIONAL CHARACTER,

    -- with size of 1 character
    CHAR_1_COL                                CHAR(1),
    CHAR_VARYING_1_COL                        CHAR VARYING(1),
    CHARACTER_1_COL                           CHARACTER(1),
    CHARACTER_VARYING_1_COL                   CHARACTER VARYING(1),
    VARCHAR_1_COL                             VARCHAR(1),
    VARCHAR2_1_COL                            VARCHAR2(1),
    NCHAR_1_COL                               NCHAR(1),
    NCHAR_VARYING_1_COL                       NCHAR VARYING(1),
    NATIONAL_CHAR_VARYING_1_COL               NATIONAL CHAR VARYING(1),
    NATIONAL_CHARACTER_VARYING_1_COL          NATIONAL CHARACTER VARYING(1),
    NVARCHAR2_1_COL                           NVARCHAR2(1),

    -- with maximum size of 2000 characters
    CHAR_2000_COL                       CHAR(2000),
    CHAR_VARYING_2000_COL               CHAR VARYING(2000),
    CHARACTER_2000_COL                  CHARACTER(2000),
    CHARACTER_VARYING_2000_COL          CHARACTER VARYING(2000),
    VARCHAR_2000_COL                    VARCHAR(2000),
    NCHAR_1000_COL                      NCHAR(1000),
    NCHAR_VARYING_2000_COL              NCHAR VARYING(2000),
    NATIONAL_CHAR_VARYING_2000_COL      NATIONAL CHAR VARYING(2000),
    NATIONAL_CHARACTER_VARYING_2000_COL NATIONAL CHARACTER VARYING(2000),

    LONG_COL                            LONG,

    /* Since LONG VARCHAR also maps to Oracle LONG, it is not possible to use
       LONG VARCHAR aside from LONG in one table. For LONG VARCHAR, see
       OTHER_DATA_TYPES.sql.

       LONG_VARCHAR_COL                          LONG VARCHAR,

     */


    -- CLOB and NCLOB have no dimensions
    CLOB_COL                                  CLOB,
    NCLOB_COL                                 NCLOB

)