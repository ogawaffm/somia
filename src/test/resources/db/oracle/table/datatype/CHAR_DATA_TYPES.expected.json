{
  "name": "CHAR_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "CHAR_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_COL",
      "type": {
        "name": "NCHAR",
        "typeNumber": -15,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR_COL",
      "type": {
        "name": "NCHAR",
        "typeNumber": -15,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER_COL",
      "type": {
        "name": "NCHAR",
        "typeNumber": -15,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_1_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_VARYING_1_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_1_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_VARYING_1_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_1_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR2_1_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_1_COL",
      "type": {
        "name": "NCHAR",
        "typeNumber": -15,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_VARYING_1_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR_VARYING_1_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER_VARYING_1_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NVARCHAR2_1_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_2000_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_VARYING_2000_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_2000_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_VARYING_2000_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_2000_COL",
      "type": {
        "name": "VARCHAR2",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_1000_COL",
      "type": {
        "name": "NCHAR",
        "typeNumber": -15,
        "className": "java.lang.String",
        "precision": 1000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_VARYING_2000_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR_VARYING_2000_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER_VARYING_2000_COL",
      "type": {
        "name": "NVARCHAR2",
        "typeNumber": -9,
        "className": "java.lang.String",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "LONG_COL",
      "type": {
        "name": "LONG",
        "typeNumber": -1,
        "className": "java.lang.String",
        "precision": 0,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 65535,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CLOB_COL",
      "type": {
        "name": "CLOB",
        "typeNumber": 2005,
        "className": "java.sql.Clob",
        "precision": 4000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 4000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCLOB_COL",
      "type": {
        "name": "NCLOB",
        "typeNumber": 2011,
        "className": "java.sql.NClob",
        "precision": 4000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 4000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}